pkg/server/plugin/zmq/broker.go

```diff
@@ -221,50 +243,93 @@ func (lb *Broker) Channeler() {
 			lb.logger.Debugf("zmq/broker: zmq => channel %#x, %s\n", frames[0], frames)
 			// Dispacth back to the channel based on the zmq first frame
 			address := string(frames[0])
-			respChan, ok := lb.addressChan[address]
-			if !ok {
-				lb.logger.Infof("zmq/broker: chan not found for worker %s\n", address)
+			messageType := string(frames[2])
+			bounceCount, err := strconv.Atoi(string(frames[3]))
+			if err != nil {
+				lb.logger.Infof("zmq/broker: Cannot parse bounce_count int %v\n", frames)
 				break
 			}
-			delete(lb.addressChan, address)
-			respChan <- frames[2]
+			requestID := string(frames[4])
+			message := frames[6]
+			if messageType == Request {
+				parcel := newParcel(message)
+				parcel.Worker = address
+				lb.ReqChan <- parcel
+				go func (p *Parcel) {
+					response := <- p.RespChan
+					frames := [][]byte{
+						[]byte(address),
+						[]byte(address),
+						[]byte{},
+						[]byte(Response),
+						[]byte(strconv.Itoa(bounceCount)),
+						[]byte(requestID),
+						[]byte{},
+						response,
+					}
+					lb.respChan <- frames
+				}(parcel)
+			} else if messageType == Response {
+				respChan, ok := lb.addressChan[requestID]
+				if !ok {
+					lb.logger.Infof("zmq/broker: chan not found for worker %s\n", address)
+					break
+				}
+				delete(lb.addressChan, requestID)
+				respChan <- message
+			}
 		case p := <-lb.recvChan:
 			// Save the chan and dispatch the message to zmq
 			// If current no worker ready, will retry after HeartbeatInterval.
 			// Retry for HeartbeatLiveness times
-			address := lb.workers.Next()
+			// TODO(b123400): instead of asking for a new worker
+			// get the current worker (if possible, e.g. when nested, we already have the worker id)
+			// Or, we can put the worker back to pool when received request from plugin? not very good?
```

Comment:
I think we only put the worker back to pool once the bounce count is back to zero? For worker not in pool, we retrieve it by `requestPrefix` for request that bounce count is not zero?